app/models/plan.rb

```diff
@@ -196,37 +196,70 @@ def get_guidance_group_options
   #  emails org admins and org contact 
   #  adds org admins to plan with the 'reviewer' Role
   def request_feedback(user)
-    val = Role.access_values_for(:reviewer, :commenter).min
-    self.feedback_requested = true
+    Plan.transaction do
+      begin
+        val = Role.access_values_for(:reviewer, :commenter).min
+        self.feedback_requested = true
     
-    # Share the plan with each org admin as the reviewer role
-    admins = user.org.org_admins
-    admins.each do |admin|
-      role = Role.new(user: admin, access: val)
-      self.roles << role unless self.users.include?(admin)
-    end 
-
-    if self.save!
-      # Send an email confirmation to the owners and co-owners
-      self.owner_and_coowners.each do |owner|
-        UserMailer.feedback_confirmation(owner, self, user).deliver_now
-      end
-  
-      # Send an email to all of the org admins as well as the Org's administrator email
-      if user.org.contact_email.present?
-        admins << User.new(email: user.org.contact_email, firstname: user.org.contact_name)
+        # Share the plan with each org admin as the reviewer role
+        admins = user.org.org_admins
+        admins.each do |admin|
+          self.roles << Role.new(user: admin, access: val)
+        end 
+
+        if self.save!
+          # Send an email confirmation to the owners and co-owners
+          deliver_if(recipients: self.owner_and_coowners, key: 'users.feedback_requested') do |r|
+            UserMailer.feedback_confirmation(r, self, user).deliver_now
+          end
+          # Send an email to all of the org admins as well as the Org's administrator email
+          if user.org.contact_email.present?
+            admins << User.new(email: user.org.contact_email, firstname: user.org.contact_name)
+          end
+          deliver_if(recipients: admins, key: 'admins.feedback_requested') do |r|
+            UserMailer.feedback_notification(r, self, user).deliver_now
+          end
+          true
+        else
+          false
+        end
+      rescue Exception => e
```

Comment:
rescue from Exception it is considered bad practice, you should handle only exceptions that you know how to recover (e.g. ActiveRecord::RecordInvalid, etc)

Reply:
Ok. we can look at readdressing later. This was intended to capture both ActiveRecord failures as well as Mailer failures, log the specific failure and rollback any DB activity that occurred so that the feedback wasn't left in a semi-complete state.